server/src/db/cidr.rs

```diff
@@ -109,14 +114,12 @@ impl DatabaseCidr {
     fn from_row(row: &rusqlite::Row) -> Result<Cidr, rusqlite::Error> {
         let id = row.get(0)?;
         let name = row.get(1)?;
-        let ip: String = row.get(2)?;
+        let ip_str: String = row.get(2)?;
         let prefix = row.get(3)?;
-        let cidr = IpNetwork::new(
-            ip.parse()
-                .map_err(|_| rusqlite::Error::ExecuteReturnedResults)?,
-            prefix,
-        )
-        .map_err(|_| rusqlite::Error::ExecuteReturnedResults)?;
+        let ip = ip_str
+            .parse()
+            .map_err(|_| rusqlite::Error::ExecuteReturnedResults)?;
+        let cidr = IpNet::new(ip, prefix).map_err(|_| rusqlite::Error::ExecuteReturnedResults)?;
```

Comment:
(preexisting)
I see that our hands are tied by https://docs.rs/rusqlite/latest/rusqlite/enum.Error.html

But it seems that there are some variants that would hopefully not be worse and would allow forwarding underlying error? I'm looking at `FromSqlConversionFailure(usize, Type, Box<dyn Error + Send + Sync + 'static>),` or `UserFunctionError(Box<dyn Error + Send + Sync + 'static>),`.

It is possible thought that visibility may not allow us to create these.

Reply:
Good point, I'll use FromSqlConversionFailure instead.